### PR TITLE
4.x Remove SimpleCacheEngine

### PR DESCRIPTION
**Pull request by markstory**
With each engine implementing the PSR16 interfaces there is no reason for this class to exist anymore. I'll be finishing off default values and TTL in set() in future pull requests.

### CODE REVIEW

**Review by ADmad — APPROVED** [review]

File: src/Cache/Cache.php [hunk 1]
```diff
@@ -223,7 +223,7 @@ public static function engine(string $config): CacheEngine
      */
     public static function pool(string $config)
     {
-        return new SimpleCacheEngine(static::engine($config));
+        return static::engine($config);
```
Comment (saeideng) [inline src/Cache/Cache.php]: `static::engine()` deprecated

Comment (markstory) [reply src/Cache/Cache.php]: Yes. That is also on my list of things to change, but I wanted to take small steps.

**Review by lorenzo — APPROVED** [review]
LGTM

File: src/Cache/Cache.php (hunk 1 above)
Comment (saeideng) [reply src/Cache/Cache.php]: so , goods for this step :+1:

**Pull request merged**